### PR TITLE
feat: nifi timestamp precision

### PR DESCRIPTION
**Pull request by helenosheaa**
helps https://github.com/influxdata/data-acquisition/issues/204
related https://github.com/influxdata/idpe/pull/15037

- adds the ability to provide a timestamp precision
- updates tests

supported timestamp types pulled from PutInfluxDatabaseRecord_2 which is using import java.util.concurrent.TimeUnit;
NS, MS, S, US.

two clarifications:
Currently you can set precision even when you don't provide a path/pattern to a timestamp should I make it disabled when not provided? 
And for line protocol it's always the default of NS.

Changes to Edit/Create:
Screen Shot 2022-06-30 at 2 52 56 PM

Screen Shot 2022-06-30 at 2 54 05 PM

Screen Shot 2022-07-01 at 1 09 49 PM

Details:
Screen Shot 2022-07-01 at 1 10 26 PM

Screen Shot 2022-07-01 at 3 11 45 PM

Screen Shot 2022-07-01 at 3 13 08 PM

### Checklist

Authors and Reviewer(s), please verify the following:

- [x] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [x] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [x] Documentation updated or issue created (provide link to issue/PR)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [ ] Feature flagged, if applicable

### CODE REVIEW

File: src/writeData/subscriptions/utils/form.ts [hunk 1]
```diff
@@ -82,6 +82,10 @@ export const sanitizeForm = (form: Subscription): Subscription => {
     delete form.brokerUsername
     delete form.brokerPassword
   }
+
+  if (form.dataFormat === 'lineprotocol') {
+    form.timestampPrecision = 'NS'
+  }
```
Comment (genehynson) [inline src/writeData/subscriptions/utils/form.ts:88]: I think we'll need to let users sending LP messages set this property too

File: src/writeData/subscriptions/components/JsonParsingForm.tsx [hunk 2]
```diff
@@ -46,6 +46,17 @@ const JsonParsingForm: FC<Props> = ({formContent, updateForm, edit}) => {
   const numberType = 'Number'
   const dataTypeList = [stringType, numberType]
   const [dataTypeM, setDataTypeM] = useState(stringType)
+  const microsecondsType = 'MS'
```
Comment (drdelambre) [inline src/writeData/subscriptions/components/JsonParsingForm.tsx:49]: if you're looking for tighter TS support for these kinds of patterns, i think this works:
``` Javascript
enum Precision {
  Millisecond = 'MS'
  Second = 'S'
  Microsecond = 'US'
  Nanosecond = 'NS'
}

const precisionList = Object.keys(Precision)
```

if not, please disregard

File: src/writeData/subscriptions/utils/form.ts (hunk 1 above)
Comment (helenosheaa) [reply src/writeData/subscriptions/utils/form.ts:88]: gotcha, adding now :)

File: src/writeData/subscriptions/components/JsonParsingForm.tsx (hunk 2 above)
Comment (genehynson) [inline src/writeData/subscriptions/components/JsonParsingForm.tsx:49]:
```suggestion
  const millisecondsType = 'MS'
```

File: src/writeData/subscriptions/components/JsonParsingForm.tsx [hunk 3]
```diff
@@ -46,6 +46,17 @@ const JsonParsingForm: FC<Props> = ({formContent, updateForm, edit}) => {
...
+  const secondsType = 'S'
+  const microseconds2Type = 'US'
```
Comment (genehynson) [inline src/writeData/subscriptions/components/JsonParsingForm.tsx:51]:
```suggestion
  const microsecondsType = 'US'
```

File: src/writeData/subscriptions/components/LineProtocolForm.tsx [hunk 4]
```diff
@@ -1,38 +1,79 @@
...
+  const microsecondsType = 'MS'
+  const secondsType = 'S'
+  const microseconds2Type = 'US'
+  const nanosecondsType = 'NS'
+  const precisionList = [
+    nanosecondsType,
+    microsecondsType,
+    secondsType,
+    microseconds2Type,
+  ]
```
Comment (genehynson) [inline src/writeData/subscriptions/components/LineProtocolForm.tsx:31]: I see we're defining this repeatedly - could it be abstracted somewhere? Maybe stored in a type within a context?

**Review by genehynson — CHANGES_REQUESTED** [review]
I think there's a typo in the precisions, left a comment

File: src/writeData/subscriptions/components/LineProtocolForm.tsx (hunk 4 above)
Comment (helenosheaa) [reply src/writeData/subscriptions/components/LineProtocolForm.tsx:31]: I'll pull it out!

File: src/writeData/subscriptions/components/JsonParsingForm.tsx (hunk 2 above)
Comment (helenosheaa) [reply src/writeData/subscriptions/components/JsonParsingForm.tsx]: opps thanks haha

**Comment by genehynson** [comment]
> Currently you can set precision even when you don't provide a path/pattern to a timestamp should I make it disabled when not provided?

I don't think this is necessary. It won't change the nifi processor's behavior at all to have this value set different when no timestmap is provided. And I think some people might be inclined to set the precisions first before typing in their jsonpath/regex on how to find it. If it was disabled, it could be more confusing.

**Comment by genehynson** [comment]
@helenosheaa could you add a screenshot of what this looks like in the "read only" state in the update screen?

**Comment by helenosheaa** [comment]
yes added more screenshots and pulled out the enum type. also realized I didn't need the setState as I can rely on the form funcs for managing the dropdown.

**Review by genehynson — APPROVED** [review]
Awesome!

**Pull request merged**